site/content/guides/claude-code-monitoring.md

```diff
@@ -0,0 +1,415 @@
+  })
+})
+});
+```
+
+Claude Code will confirm before creating the script file, respectively. If you've followed these tutorial steps in a single conversation with Claude Code, the tool will likely offer to `create signup-form-elements.check.ts` to configure the check.
+
+You can prompt for multiple checks at the same time. As you can imagine, this unlocks the creation of a large number of synthetic monitors relatively quickly. 
+
+```{title="Prompt:"}
+Load the page https://danube-web.shop/ and create several separate .spec.ts checks, one for each page linked off of danube-web.shop. Each created check should check for multiple page elements.
+```
+If you haven't watched the responses from Claude Code already, it can be helpful to see how the tool breaks down our request:
+
+```{title="Reply from Claude Code:"}
+⏺ Update Todos
+  ⎿  ☐ Load danube-web.shop and identify all linked pages           
+     ☐ Explore each linked page to understand available elements
+     ☐ Create .spec.ts files for each linked page with multiple
+       element checks
+```
+Every file output from this command isn't included here for brevity, but Claude Code did create five new check scripts based on the five heading links at danube-web.shop. To test all these new checks quickly, run
+
+```bash
+npx checkly test
+```
+*With no file name argument, this will try to run everything in the `__checks__` folder*
+
+```bash
+2 failed, 5 passed, 7 total
+```
+*If you've followed every step in this tutorial, you may have different numbers for the number of tests run here.*
+
+
+When running a number of generated tests, the only issue we ran into was a failure with the message: `Error: expect.toBeVisible: Error: strict mode violation: getByText('Crime & Thrillers') resolved to 2 elements:`. As the error message implies, the issue is that this locator:
+
+```ts
+  await expect(page.getByText('Fantasy')).toBeVisible();
+```
+
+Found multiple results on the page. There are a number of ways to fix this with Claude Code. We could prompt Claude Code with:
+
+```{title="Prompt:"}
+revise locators in item-visibility.spec.ts to look at only the first result
+```
+
+This would change every locator, which may or may not be what you want! You could also paste in the error message to Claude and ask it to fix the script. Whatever route you take the result will be an edit to:
+
+```ts
+  await expect(page.getByText('Fantasy').first()).toBeVisible();
+```
+If you're having trouble reading the generated tests, take a look at [Checkly's Playwright documentation](https://www.checklyhq.com/learn/playwright/). Once Claude Code added a single `.first()` to the checks written above, everything was passing. This issue proved frequent enough that we added the following to Claude Code's permanent memory:
+
+```{title="Prompt:"}
+# When writing a Playwright check, if you find multiple elements that match a locator, make sure the locator includes .first() to take the first result.
```

Comment:
This is a very **bold** statement to have in context. I'm not sure if i'd feel comfortable recommending this, because it'll drive you nuts if you don't want to select the first element. :D 🫣